S3FileRenamer.py

```diff
@@ -6,35 +6,48 @@
     Default behavior is to only generate the log file for review.  Pass the "--rename" or "-r" flag to rename files.
  """
 
-from os import walk, rename
-from os.path import join
+import os
 import argparse
 import re
 
-
-def contains_bad_characters(string):
+def contains_bad_characters(string, badchars):
     """Search input string for illegal characters.  Return True, list of characters found, if any found."""
     found_bad_characters = []
     for char in string:
-        if char in bad_characters:
+        if char in badchars:
             found_bad_characters.append(char)
     if len(found_bad_characters) > 0:
         return True, found_bad_characters
     else:
         return False, ""
 
-
```

Comment:
Python PEP 8 calls for 2 blank lines between top-level functions.  See https://www.python.org/dev/peps/pep-0008/#blank-lines  Please add back the blank lines.